meetings/2022-02-14.md

```diff
@@ -0,0 +1,112 @@
+* EP: Note that ACP also allows to restrict issuers when accessing some data. If the user is logged in multiple trusted OP, then it may have multiple ID token to select from.
+* NAS: Currently browser based clients wouldn't support being logged into multiple OP at once. We only have full window redirect, and it would require popup redirect. It could be supported on the server side but we haven't actively studied it. We'll have to check how it applies.
+
+### Authorization server behaviour for Client-Credentials Grant flow
+
+[Issue#75](https://github.com/solid/solid-oidc/issues/75)
+
+* LD: Context: in the Flamish gov, we have two different Identity Providers. A user-facing OP, and an OAuth 2.0 AuthN server for server-to-server communication. So far we are relying on the access token issued by the OP, and we used the OAuth 2.0 non-opaque access token in that role. Moving away from this pattern questions how to support this server-to-server communication. 
+* EP: In your use case, one of the server still acts as a client, correct ?
+* LD: Correct. The server app acts on behalf of an organization through the client credentials flow.
+* EP: In this flow, when you use the OAuth 2.0 access token, is there still a phase of issuer discovery ?
+* LD: The issuer for this organizational WebID is the domain for both the OP and OAuth issuer. So yes for discovery.
+* AC: One thing to mention, client credentials is more an oauth2 mechanism than oidc. It's unclear how Solid-OIDC will relate to client credentials. It is however very relevant to this panel, and we are looking at producing a different document with recommandations for this use case. However, it not being defined within OIDC doesn't prevent from using the client credentials flow with an OP. The question the becomes, how do you pair the obtained client credentials with a WebID ? There are multiple patterns: (1) using a WebID as the Client ID. There are challenges with this, in particular the client ID will be unique, and then only one client secret will be supported. (2) having an interface where the client may register client credentials. You login through a web page, and you can obtain server-managed client IDs and secrets associated to this session.
+* EP: Agreed in general. It would be worth defining some patterns for client credentials, if not specifying about it. At the end, we want a dpop-bound ID token to be obtained, so it sort of relates to the Solid-OIDC spec. If you could get an access token with the appropriate org WebID from a client credentials pair, would it work for your use case ?
+* LD: I believe so. The only major concerns from ID team is that adding the client credentials flow will fail the OIDC conformance tests, because these flows are indeed undefined in the OIDC spec.
```

Comment:
```suggestion
* EP: Agreed in general. It would be worth defining some patterns for client credentials, if not specifying about it. At the end, we want a dpop-bound ID token to be obtained, so it sort of relates to the Solid-OIDC spec. If you could get an access token with the appropriate org WebID from a client credentials pair, would it work for your use case?
* LD: I believe so. The only major concern from ID team is that adding the client credentials flow will fail the OIDC conformance tests, because these flows are indeed undefined in the OIDC spec.
```